gooddata-sdk/gooddata_sdk/catalog.py

```diff
@@ -17,99 +17,55 @@
-        raise NotImplementedError()
+        return self._entity["id"]
 
     @property
     def type(self) -> str:
-        raise NotImplementedError()
-
-    @property
-    def obj_id(self) -> ObjId:
-        raise NotImplementedError()
+        return self._entity["type"]
 
     @property
     def title(self) -> str:
-        raise NotImplementedError()
+        return self._e.get("title")
 
     @property
     def description(self) -> str:
-        raise NotImplementedError()
-
-
-class CatalogLabel(CatalogEntry):
-    def __init__(self, label: dict[str, Any]) -> None:
-        super(CatalogLabel, self).__init__()
-
-        self._l = label["attributes"]
-        self._label = label
-        self._obj_id = ObjId(self._label["id"], type=self._label["type"])
-
-    @property
-    def id(self) -> str:
-        return self._label["id"]
-
-    @property
-    def type(self) -> str:
-        return self._label["type"]
-
-    @property
-    def title(self) -> str:
-        return self._label["type"]
+        return self._e.get("description")
 
     @property
     def obj_id(self) -> ObjId:
         return self._obj_id
 
-    @property
-    def description(self) -> str:
-        return self._l["description"]
+    def __str__(self) -> str:
+        return self.__repr__()
 
+    def __repr__(self) -> str:
+        return f"CatalogFact(id={self.id}, title={self.title})"
+
+
+class CatalogLabel(CatalogEntry):
     @property
     def primary(self) -> bool:
-        return "primary" in self._l and self._l["primary"]
+        return "primary" in self._e and self._e["primary"]
 
     def as_computable(self) -> Attribute:
         return Attribute(local_id=self.id, label=self.id)
 
-    def __str__(self) -> str:
-        return self.__repr__()
-
-    def __repr__(self) -> str:
-        return f"CatalogLabel(id={self.id}, title={self.title})"
-
 
 class CatalogAttribute(CatalogEntry):
-    def __init__(self, attribute: dict[str, Any], labels: list[CatalogLabel]) -> None:
-        super(CatalogAttribute, self).__init__()
+    def __init__(self, entity: dict[str, Any], labels: list[CatalogLabel]) -> None:
+        super(CatalogAttribute, self).__init__(entity)
 
-        self._a = attribute["attributes"]
-        self._attribute = attribute
         self._labels = labels
         self._labels_idx = dict([(str(label.obj_id), label) for label in labels])
         self._dataset: Optional[CatalogDataset] = None
-        self._obj_id = ObjId(self._attribute["id"], self._attribute["type"])
-
-    @property
-    def id(self) -> str:
-        return self._attribute["id"]
-
-    @property
-    def type(self) -> str:
-        return self._attribute["type"]
-
-    @property
-    def obj_id(self) -> ObjId:
-        return self._obj_id
-
-    @property
-    def title(self) -> str:
-        return self._a["title"]
-
-    @property
-    def description(self) -> str:
-        return self._a["description"]
+        self._obj_id = ObjId(self._entity["id"], self._entity["type"])
```

Comment:
Do we need to create self._obj_id? It's created in parents constructor.

Reply:
removed, letover.